### ISSUE TITLE
OpenDialog unexpected behavior in Ubuntu's 'Recently Used' area of file picker

### ISSUE THREAD

**Issue by lbell** [electron/electron#10197]
* Electron version: 1.4.15
* Operating system: Ubuntu 16.04

### Expected behavior

```
dialog.showOpenDialog({
      properties: [
        'openFile',
      ],
    })
```
Should allow you to only open files... which it does, most of the time, and will not proceed if you've accidentally selected a directory.

### Actual behavior

If you go into Ubuntu's "Recently Used" directory in the file picker window, then if you select a directory, electron will gladly return the directory.

Sidenote: if you have a filter set up -- say for extensions: ['jpg'] -- electron returns the directory path with .jpg appended.

**Comment by sofianguy** [comment]
Thanks for contributing!

We are no longer releasing bug fixes for versions of Electron older than 1.5. We prefer to only keep issues that are on our immediate radar.

If you are still having the same issue in an up-to-date version, feel free to reach out and we'll reopen.

**Comment by ntpincus** [comment]
I am experiencing this bug on electron v2.0.1 except with Linux
